lib/rust/flexer/src/automata/nfa.rs

```diff
@@ -200,14 +208,14 @@ impl From<&NFA> for DFA {
         let mut callbacks = vec![None; dfa_eps_ixs.len()];
         let     priority  = dfa_eps_ixs.len();
         for (dfa_ix, epss) in dfa_eps_ixs.into_iter().enumerate() {
-            let has_name = |&key:&state::Id| nfa.states[key.id].name.is_some();
+            let has_name = |&key:&state::Identifier| nfa.states[key.id].name.is_some();
             if let Some(eps) = epss.into_iter().find(has_name) {
                 let rule  = nfa.states[eps.id].name.as_ref().cloned().unwrap();
                 callbacks[dfa_ix] = Some(Callback {name:rule,priority});
             }
         }
 
-        DFA {alphabet:nfa.alphabet.clone(),links:dfa_mat,callbacks}
+        DFA { alphabet_segmentation:nfa.alphabet.clone(),links:dfa_mat,callbacks}
```

Comment:
please, create separate variables for `alphabet_segmentation` and `links` and then use simple constructor form as `Self {alphabet_segmentation,links,callbacks}` instead